### PR TITLE
Add get sql function

### LINKED ISSUE #53: Add contributing guidelines
Would be helpful to have a dedicated contributing page that pulls from our knowledge and the [R Package (2e) book by Hadley Wickham and Jennifer Bryan](https://r-pkgs.org/) that lowers the bar for more DfE analysts to be able to contribute to this.

Some quick initial thoughts:

- Standard workflows for running the package locally with `devtools` and `usethis`
- Detailing the purpose for the code
- Adding suitable documentation using `roxygen`
- Adding suitable test coverage using `testthat`
- Ensuring that code meets the `tidyverse` styling convention
- Get approving review from at least one of the maintainers

### LINKED ISSUE #55: Add in read_sql() type function
There were sql_conn_string() and read_sql_script() in the old package, my team has updated code for these that should be added, including read_sql(), where it automatically cleans a .sql script to read in via R, dodging common pitfalls.

Should also make the function add the two lines that we usually recommend adding to the top of SQL scripts that are read by R:

```
 SET ANSI_PADDING OFF
 SET NOCOUNT ON;
```

### PR DESCRIPTION
**Pull request by cjrace**
# Brief overview of changes

Add a get_clean_sql function to resolve #55, and details on code coverage setup to resolve #53.

This increases the package version to 0.2.0.

## Why are these changes being made?

This function is used across lots of DfE projects already. Other bits of tidy up to resolve known issues with documentation.

## Detailed description of changes

Updates to contributing page:
- Typo fix on `usethis::use_test()`
- Extra information on TDD approach and roxygen2 documentation
- Typo fix on `lintr::lint_package()`
- Reformat keyboard shortcuts section so it's easier to see
- Add details on test coverage checks
- Add instructions for adding a new vignette

Updates to package documentation:
- Added titles into _pkgdown.yml to group functions in reference list

Added `get_clean_sql()` function, documentation and tests. Includes a sample SQL script to test with.

Added vignette for connecting to SQL to give more user friendly context for `get_clean_sql()`.

Bumping package version to 0.2.0 for this and #59.

Added note to README pointing to the reference documentation for the package.

## Additional information for reviewers

1. Use the following line to preview package documentation updates locally:

```
devtools::build_site()
```

2. I will squash the commits when merging as I got into a slight mess with git rebase conflicts that seems to have duplicated commits in the commit history.

3. I'll reflect any changes here in https://github.com/dfe-analytical-services/analysts-guide/pull/43 to update the relevant section of the Analysts' Guide.

## Issue ticket number/s and link

#55, #53.

### CODE REVIEW

**Comment by cjrace** [comment]
[Add SQL functions into dfeR](https://trello.com/c/Z1dOmDsg/1145-add-sql-functions-into-dfer)

File: vignettes/connecting_to_sql.Rmd [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+## Executing the SQL query
+
+Now that the SQL query is saved as a variable in the R environment you can pass that into a function to execute against the database. There's a number of potential ways to do this, though a common way is to use `dbGetQuery()` from the [DBI package](https://dbi.r-dbi.org/), setting the statement as your cleaned SQL query.
+
+```{r executing sql query, eval=FALSE}
+sql_query_result <- DBI::dbGetQuery(con, statement = sql_query)
+```
+
```
Comment (rmbielby) [inline vignettes/connecting_to_sql.Rmd:68]: Are there any common things that won't get cleaned by the script and won't run with `dbGetQuery()`?
e.g. will something like the following run:
```
USE [SWFC_Project_Restricted]
GO
```

Worth having a section here to collate those potential trip-hazards.

File: DESCRIPTION [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Type: Package
 Package: dfeR
 Title: Common DfE R tasks
-Version: 0.1.1
+Version: 0.2.0
 Authors@R: c(
     person("Cam", "Race", , "cameron.race@education.gov.uk", role = c("aut", "cre")),
     person("Laura", "Selby", , "laura.selby@education.gov.uk", role = "aut"),
```
Comment (rmbielby) [inline DESCRIPTION:7]: Think Jen worked on the `format_ay()` so could do with adding her as a contributor?

File: vignettes/connecting_to_sql.Rmd [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+)
+```
+
+R can be used to execute SQL scripts to extract data from a database as well as querying the database directly via R. There are three primary ways to do this:
+
+1. executing a separate SQL script from R
+2. writing strings of SQL code in your R script
+3. using [dbplyr](https://dbplyr.tidyverse.org/) to query a database using R code
+
+Which you use will depend on how comfortable with SQL and R and also if you already have existing SQL scripts that you want to execute or you’re writing new database queries. This vignette focuses on that first example, using the `get_clean_sql()` function to read in a separate SQL script and execute from R.
```
Comment (rmbielby) [inline vignettes/connecting_to_sql.Rmd:23]: ...how comfortable _you are_ with...

File: vignettes/connecting_to_sql.Rmd [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+For more information on the other two methods, or on troubleshooting the connection between R and SQL in the Department for Education (DfE), please see the [Interacting with a SQL database section of our Analysts' Guide](https://dfe-analytical-services.github.io/analysts-guide/learning-development/r.html#interacting-with-a-sql-database-from-within-r-scripts).
+
+## Connecting to a database
+
+Usually in the DfE we use a combination of [odbc](https://odbc.r-dbi.org/) and [DBI](https://dbi.r-dbi.org/) to connect to our SQL databases.
+
+How you connect will vary depending on whether you're running R code on your laptop, or as a part of a deployed R Shiny, for running code on your laptop you can automatically use your Windows login (a trusted connection) to grant you access to the database, as the package can automatically detect your user details.
```
Comment (rmbielby) [inline vignettes/connecting_to_sql.Rmd:31]: or as part of a deployed R Shiny _app_,

Comment (rmbielby) [inline vignettes/connecting_to_sql.Rmd:31]: Could do with splitting into two sentences. Start a new sentence after "R Shiny _app_" maybe

File: vignettes/connecting_to_sql.Rmd [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  PWD = "",
+  Trusted_Connection = "Yes"
+)
+```
+
+For advice on finding your database details, or connecting to a SQL database from an R Shiny app that is deployed on a server, please contact the [Statistics Development Team](mailto:statistics.development@education.gov.uk) who will be able to advise on the setup and steps required.
+
+## Reading a SQL script into R
+
+There are a number of standard characters found in SQL scripts that can cause issues when reading in a SQL script within R, we have created the `get_clean_sql()` function to assist with this. Assume you have connected to the database and assigned that connection to a `con` variable, you would then use the following line to read a cleaned version of your SQL script into R.
```
Comment (rmbielby) [inline vignettes/connecting_to_sql.Rmd:55]: -> "in a SQL script within R _and_ the `get_clean_sql()` function has been created to assist with this."

**Review by rmbielby — CHANGES_REQUESTED** [review]
Mainly just a few minor tweaks to the text.

File: vignettes/connecting_to_sql.Rmd (hunk 1 above)
Comment (cjrace) [reply vignettes/connecting_to_sql.Rmd:70]: Resolved(ish) by specifying that these are only intended to work with 'SELECT' style statements, with the aim of expanding on this as we go (expecting there'll be additions we can make for anything around connecting to Databricks, and other servers as more people start using it)

**Review by rmbielby — APPROVED** [review]

**Pull request merged**